pilot/docker/prepare_proxy.sh

```diff
@@ -6,16 +6,18 @@ set -o nounset
 set -o pipefail
 
 usage() {
-  echo "${0} -p PORT -u UID [-h]"
+  echo "${0} -p PORT -u UID [-g] [-h]"
   echo ''
   echo '  -p: Specify the envoy port to which redirect all TCP traffic'
   echo '  -u: Specify the UID of the user for which the redirection is not'
   echo '      applied. Typically, this is the UID of the proxy container'
+  echo '  -g: Specify configuration for a gateway, in which case inbound traffic will not be redirected'
   echo '  -i: Comma separated list of IP ranges in CIDR form to redirect to envoy (optional)'
   echo ''
 }
 
 IP_RANGES_INCLUDE=""
+REDIRECT_INBOUND_TRAFFIC="yes" # Route all inbound traffic to Envoy by default
```

Comment:
nit: TRAP_INBOUND_TRAFFIC and change the rest to reflect this statement.. In general, we want TRAP_INBOUND, TRAP_OUTBOUND, etc.. If at all possible, use a different flag than -g.
And this does not have much to do with gateway btw, as gateways will not have any IPTables stuff.